p2p/raintree/network.go

```diff
@@ -179,6 +206,12 @@ func (n *rainTreeNetwork) HandleNetworkData(data []byte) ([]byte, error) {
 	}
 
 	n.mempool[rainTreeMsg.Nonce] = struct{}{}
+	n.mempool_keys = append(n.mempool_keys, rainTreeMsg.Nonce)
+	if len(n.mempool_keys) > mempoolMaxNonces {
```

Comment:
What are your thoughts on potentially generalizing and reusing the mempool from `utility/types/mempool.go`?

It would require:

1. Me merging my PR
2. Moving mempool into general
3. Generalizing some names (e.g. Tx to a generic)
4. Modifying the appropriate call sites

<img width="821" alt="Screenshot 2022-12-06 at 5 21 39 PM" src="https://user-images.githubusercontent.com/1892194/206064608-14acdec7-4753-4f69-a837-55f0ba6c5fce.png">

Definitely another ticket but wanted to ask you first.

I think the terms `nonceSet` and `nonceList` would be more semantic here.

Reply:
I totally agree, we should generalize the concept and apply it wherever needed. 

Adding a TODO (lemme know if I should create the issue as well or if this should be considered as part of the discussion in #278).

I am just renaming to `nonceSet` and `nonceList` for now.

Reply:
I created #388 to track this and assigned it to you as I had to fix one last unit test in #370.

Reply:
@deblasis #388 should be done separately but will really help with both this and utility.

Reply:
> @deblasis #388 should be done separately but will really help with both this and utility.

Are you saying that #388 is now a blocker for #271?

Reply:
Nope. We created the ticket and I'm leaving a suggestion to include it in the comment in the code.